velox/common/base/Scratch.h

```diff
@@ -69,7 +69,7 @@ class Scratch {
   }
 
  private:
-  void reserve(int32_t newCapacity) {
+  void reserve(uint32_t newCapacity) {
```

Comment:
I think we just need a hint here for the compiler to know `capacity_` is always non-negative.  Just add `folly::assume(capacity_ >= 0)` is enough.

Reply:
Sure